ui/helpers/utils/util.js

```diff
@@ -702,7 +702,7 @@ export const hexToText = (hex) => {
   try {
     const stripped = stripHexPrefix(hex);
     const buff = Buffer.from(stripped, 'hex');
-    return buff.length === 32 ? hex : buff.toString('utf8');
+    return buff.toString('utf8');
```

Comment:
I am not sure why this was added in first place `buff.length === 32`. But what I can confirm is that currently this fn is used for only personal signs.

Reply:
I believe this was an intentional design decision so if a dApp provides a 32 byte buffer, we assume it's a hash and render it as hex as the UTF8 equivalent would be unreadable.

So much so that we had a bug in mobile a few months back and I replicated it there for consistency, as one of the SDK users wanted to provide a hash but it was rendering incorrectly.

Reply:
Thanks a lot @matthewwalsh0 , this is very useful point.

I am not sure now how should we fix this issue, @bschorchit can you plz suggest.

Reply:
Is there a different way we can check if it's a hex or not?

Reply:
Can the API be refactored in order to make the input not ambiguous in the first place? It seems that there is some accounting for double-encoding and that this should either be done consistently or not at all for each call-path?

Reply:
Personal sign api is defined by EIP, I do not think it will be possible to change that

Reply:
> Personal sign api is defined by EIP, I do not think it will be possible to change that

@jpuri Ah... Would you be able to point to the relevant EIP section? :pray: